omc3/optics_measurements/crdt.py

```diff
@@ -0,0 +1,228 @@
+        cols.remove(f"AMP{plane}")
+        lin_df[f"AMP{plane}"] = lin_df.loc[:, f"AMP{plane}"].to_numpy()/2.
+        lin_df.loc[:, cols] = lin_df.loc[:, cols].mul(lin_df.loc[:, f'AMP{plane}'], axis="index")
+        processed_files.append(lin_df)
+    return processed_files
+
+def process_crdt(joined_df, crdt):
+    df = pd.DataFrame(index=joined_df.index, data={'S': joined_df['S']})
+
+    lines = {}
+    phases = {}
+    errlines = {}
+    errphases = {}
+    for data_dict, prefix in zip([lines, phases, errlines, errphases], ['AMP', 'FREQ', 'ERRAMP', 'ERRFREQ']):
+        for key, line in crdt['lines'].items():
+            try:
+                translate_line = translate_line_to_col(line)
+                data_dict[key] = joined_df[translate_line[prefix]]
+            except KeyError:
+                LOGGER.debug(f"No {prefix} for line {line} found in lin-files, set to 0")
+                data_dict[key] = np.nan
+    df = crdt['func'](df, lines, phases, errlines, errphases, crdt['sign'])
+    return df
+
+
+def average_results(result_dfs, union_columns, merge_columns, merge_functions, index):
+    assert len(merge_columns) == len(merge_functions)
+
+    result_df = reduce(lambda left, right: pd.merge(left.reset_index(level=0)[union_columns],
+                                                    right.reset_index(level=0)[union_columns],
+                                                    on=union_columns,
+                                                    how='outer').set_index(index),
+                       result_dfs)
+    result_dfs = [df.reindex(result_df.index) for df in result_dfs]
+    for column, func in zip(merge_columns, merge_functions):
+        data = np.array([df[column].to_numpy() for df in result_dfs])
+        result_df[column] = func(data, axis=0)
+    return result_df
+
+
+def translate_line_to_col(line):
+    plane = line[0]
+    line = line[1:]
+
+    if (plane == 'X' and line == '10') or (plane == 'Y' and line == '01'):
+        return {'AMP': f'AMP{plane}', 'FREQ': f'MU{plane}', 'ERRAMP': f'ERRAMP{plane}', 'ERRFREQ': f'ERRMU{plane}'}
+    return {'AMP': f'AMP{line}_{plane}', 'FREQ': f'PHASE{line}_{plane}', 'ERRAMP': f'ERRAMP{line}_{plane}', 'ERRFREQ': f'ERRPHASE{line}_{plane}'}
+
+
+def write(df, header, meas_input, order, crdt):
+    outputdir = Path(meas_input.outputdir)/"crdt"/order
+    iotools.create_dirs(outputdir)
+    tfs.write(str(outputdir/f"{crdt}{EXT}"), df, header, save_index='NAME')
+
+def joined_planes(input_files):
+    """
+    Merges DataFrame from the two planes in one df
+    Parameters:
+        input_files
+    Returns:
+        merged DataFrame 
+    """
+    joined_dfs = []
+
+    assert len(input_files['X']) == len(input_files['Y'])
+
+    for linx, liny in zip(input_files['X'], input_files['Y']):
+
+        for df, plane in zip((linx, liny), PLANES):
+            rename_cols(df, plane, ['NAME', 'S', f'TUNE{plane}', f'AMP{plane}', f'MU{plane}', f'MU{plane}SYNC'])
```

Comment:
Constants

Reply:
fixed .. also in harpy, optics meas still missing